tests/sentry/stacktraces/test_functions.py

```diff
@@ -109,6 +109,40 @@ def test_trim_csharp_function_name(input, output):
     assert trim_function_name(input, "csharp") == output
 
 
+@pytest.mark.parametrize(
+    "input,output",
+    [
+        [
+            "thunk for @escaping @callee_guaranteed () -> ()",
+            "thunk/partial apply for anonymous function",
+        ],
+        [
+            "specialized thunk for @callee_guaranteed (@guaranteed [T1]) -> (@owned [T2])",
+            "thunk/partial apply for anonymous function",
+        ],
+        [
+            "partial apply for thunk for @callee_guaranteed () -> (@error @owned Error)",
+            "thunk/partial apply for anonymous function",
+        ],
+        [
+            "partial apply for thunk for @escaping @callee_guaranteed (@guaranteed SomeType, @guaranteed [String : SomeType2], @guaranteed SomeType3) -> ()",
+            "thunk/partial apply for anonymous function",
+        ],
+        [  # "closure ... in ..." functions are converted to containing
+            # function. We might want to change this in the future
+            "closure #1 (T1) in foo(bar: T2)",
+            "foo",
```

Comment:
@untitaker This stays the way it was, but it might be worth opening another PR which replaces "foo" by "closure" in these cases.